config/jobs/cert-manager/cert-manager-periodics.yaml

```diff
@@ -521,13 +521,14 @@ periodics:
         cd test-infra/aws && \
         terraform init && \
         terraform apply -auto-approve && \
+        trap 'terraform destroy -auto-approve' ERR && \
```

Comment:
Suggestion: move the trap to line 522 - I think it needs to be defined before running `terraform apply` otherwise it won't run if `terraform apply` exits. (As I understand it works similarly to `defer` statement in Go if that helps).